### PR TITLE
#65 Resolve NPE when calling generation code directly outside of a Ma…

### LINKED ISSUE #65: BUG: Calling GenerateSourcesHelper.performSourceGeneration(...) on a deprecated profile outside of a maven build causes a null pointer exception
When performing source generation on a deprecated profile from outside of a Maven build session (ie: calling the generation function directly in a unit test), a null pointer exception occurs as the PriorityMessage Service cannot find a Maven Session to attach the output to.

### PR DESCRIPTION
**Pull request by peter-mcclonski**
…ven session.

### CODE REVIEW

File: fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/util/PriorityMessageService.java [hunk 1]
```diff
@@ -73,6 +73,8 @@ public static List<PriorityMessage> getPriorityMessages(final MavenSession sessi
             }
         } catch (JsonProcessingException e) {
             logger.error("Unable to get priority messages", e);
+        } catch (NullPointerException e) {
+            logger.error("No active Maven session available!", e);
```
Comment (peter-mcclonski) [inline fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/util/PriorityMessageService.java:79]: I: I more or less defaulted to treating this case as a no-op, as it's most likely to be met in a third party unit testing situation where a no-op makes sense.  With that said, I'm open to discussion of the best path to take here.

**Comment by d-ryan-ashcraft** [comment]
S (Substantive): this is probably fine, but I'm not sure we should support functionality without a Maven Session as that is a fairly critical part of fermenter-mda and the message service work.  What is the use case?

**Comment by peter-mcclonski** [comment]
> S (Substantive): this is probably fine, but I'm not sure we should support functionality without a Maven Session as that is a fairly critical part of fermenter-mda and the message service work.  What is the use case?

We've seen cases where users of fermenter will call the generation code directly as part of a unit test. For instance, verifying that a set of project-defined profiles will generate. In that case, even though it may technically be part of a Maven build, the method of calling fermenter generation still results in the session being null.

**Comment by d-ryan-ashcraft** [comment]
> > S (Substantive): this is probably fine, but I'm not sure we should support functionality without a Maven Session as that is a fairly critical part of fermenter-mda and the message service work.  What is the use case?
> 
> We've seen cases where users of fermenter will call the generation code directly as part of a unit test. For instance, verifying that a set of project-defined profiles will generate. In that case, even though it may technically be part of a Maven build, the method of calling fermenter generation still results in the session being null.

I'm happy to go w/ maintainer consensus here.  I'm going to vote "no" and here is why:  
- While a small change and convenient, it seems fundamentally incorrect to just catch and log errors in a class that is _centered_ around the Maven Session to handle tests written in a way we don't recommend
- There is a test harness and approach to help w/ these use cases and I think we should hold the line folks using those

All that said, let's see where the consensus takes us.  Should have an decision early this AM.

**Comment by epwilkins** [comment]
I strongly concur with Ryan.  This isn't a usage pattern that Fermenter is meant to support, and adding it for unit testing seems like the wrong way to solve the unit testing issue.

**Comment by peter-mcclonski** [comment]
Discussed more offline.  Closing this PR, but keeping the issue open.  The future fix will _likely_ involve wholesale removal of the Priority Message Service, and instead adding its usages to the Notification Service.

**Comment by ekoniec1** [comment]
Given that `PriorityMessageService` is constructed with a `MavenSession`, I thin it's a very reasonable pre-condition that in order to use `PriorityMessageService`, you need a valid `MavenSession`.  IMO failing loudly is better here than catching and logging, which will likely go unnoticed by most developers.  It is a bit more work, but recommend re-thinking the approach to unit testing in this scenario.

**Pull request closed without merging**